src/components/SearchableDropdown/index.tsx

```diff
@@ -0,0 +1,237 @@
+import { useState, PropsWithChildren, useRef, useEffect } from "react";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { uiColors } from "@leafygreen-ui/palette";
+import { Body, Label } from "@leafygreen-ui/typography";
+import { Input } from "antd";
+import Icon from "components/Icon";
+import { toggleArray } from "utils/array";
+
+const { Search } = Input;
+const { gray, white, blue } = uiColors;
+
+interface SearchableDropdownProps<T> {
+  label: string | React.ReactNode;
+  value: string | T | string[] | T[];
+  onChange: (value: string | T | string[] | T[]) => void;
+  searchFunc?: (
+    value: string | T | string[] | T[],
+    match: string | T
+  ) => boolean;
+  searchPlaceholder?: string;
+  options: string[] | Array<T>;
+  optionRenderer?: (option: string | T) => React.ReactNode;
+  allowMultiselect?: boolean;
+}
+const SearchableDropdown = <T extends {}>({
+  label,
+  value,
+  onChange,
+  searchFunc,
+  searchPlaceholder = "search...",
+  options,
+  optionRenderer,
+  allowMultiselect = false,
+}: PropsWithChildren<SearchableDropdownProps<T>>) => {
+  const [isOpen, setisOpen] = useState(false);
+  const [search, setSearch] = useState("");
+  const [visibleOptions, setVisibleOptions] = useState(options);
+
+  const listMenuRef = useRef(null);
+  const menuButtonRef = useRef(null);
+
+  // Handle onClickOutside
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+    const onClickOutside = (event: MouseEvent) => {
+      const stillFocused =
+        menuButtonRef.current!.contains(event.target as Node) ||
+        listMenuRef.current!.contains(event.target as Node);
+      setisOpen(stillFocused);
+    };
+
+    document.addEventListener("mousedown", onClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", onClickOutside);
+    };
+  }, [listMenuRef, menuButtonRef, isOpen]);
+
+  const onClick = (v: string | T) => {
+    if (allowMultiselect) {
+      if (Array.isArray(value)) {
+        const newValue = toggleArray(v, value) as any[];
+        onChange(newValue);
+      }
+    } else {
+      onChange(v);
+    }
+    // Close the dropdown after user makes a selection only if it isn't a multiselect
+    if (!allowMultiselect) {
+      setisOpen(false);
+    }
+  };
+
+  const option =
+    optionRenderer ||
+    ((v: string | T) => (
+      <Option
+        onClick={() => onClick(v)}
+        key={`select_${v}`}
+        data-cy="searchable-dropdown-option"
+      >
+        <CheckmarkContainer>
+          {isChecked(v) && (
+            <Icon glyph="Checkmark" height={12} width={12} fill={blue.base} />
+          )}
+        </CheckmarkContainer>
```

Comment:
Right now multi-select is slightly hidden from the user imo -- there's no visual difference between the single and multi select dropdown when it's initially opened. Maybe including empty checkmark boxes when `isChecked` is false would make this more apparent?

Reply:
![image](https://user-images.githubusercontent.com/4605522/129931050-19ae6179-5042-47ac-ad0d-4c04800844ee.png)
I based it off of the designs used here.